base/MAPL_CubedSphereGridFactory.F90

```diff
@@ -723,10 +771,65 @@ elemental subroutine set_with_default_bounds(to, from, default)
       
    end subroutine set_with_default_bounds
    
+   logical function test_decomp_equals(this, a)
```

Comment:
I would prefer a name that does not start with `test_` which is the semaphore use for unit tests.  How about just `decoms_equal` or `equal_decomps`?

Reply:
NP, I will change that

Reply:
Better suggestion is `decomps_are_equal` as per above.